### PR TITLE
Enum / IT-Academy / JC1-117-22-dev / Haretski Aliaksei / Task01-07

### PR DESCRIPTION
**Pull request by haretski-aliaksei**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

**Comment by haretski-aliaksei** [comment]
Previous PR I put in incorrect section of "Rakovets Lab" (lesson09 < lesson10):
https://github.com/rakovets/course-java-core/pull/986

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        @Override
+        public String getDescription() {
+            return "Warm season";
+        }
+
+        @Override
+        public String toString() {
+            String summerToString = String.format("%s %,.1f %s", SUMMER.name(), SUMMER.getAverageTemperaturePerSeason(SUMMER), SUMMER.getDescription());
+            return summerToString;
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java:28]: move to enum, not INSTANCE

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package com.rakovets.course.java.core.practice.enum_types;
+
+public enum Task01040506Seasons {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java:3]:
```suggestion
public enum Season {
```

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    },
+    WINTER(-25.0) {
+        @Override
+        public String toString() {
+            String winterToString = String.format("%s %,.1f %s", WINTER.name(), WINTER.getAverageTemperaturePerSeason(WINTER), WINTER.getDescription());
+            return winterToString;
+        }
+    };
+
+    private double averageTemperaturePerSeason;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java:38]:
```suggestion
    private final double averageTemperaturePerSeason;
```

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    Task01040506Seasons(double averageTemperature) {
+        this.averageTemperaturePerSeason = averageTemperature;
+    }
+
+    public String getDescription() {
+        return "Cold season";
+    }
+
+    public double getAverageTemperaturePerSeason(Task01040506Seasons season) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/Task01040506Seasons.java:48]:
```suggestion
    public double getAverageTemperature(Task01040506Seasons season) {
```

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/Task03SeasonsSwitch.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.rakovets.course.java.core.practice.enum_types;
+
+public class Task03SeasonsSwitch {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/Task03SeasonsSwitch.java:3]: move to `test`

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/Task07Print.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.rakovets.course.java.core.practice.enum_types;
+
+public class Task07Print {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/Task07Print.java:3]: move to `test`

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/Task02PrintEnum.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package com.rakovets.course.java.core.practice.enum_types;
+
+public class Task02PrintEnum {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/Task02PrintEnum.java:3]: move to `test`

File: src/test/java/com/rakovets/course/java/core/practice/enum_types/TestEnum.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideArgumentsTestGetAverageTemperaturePerSeason")
+    public void testGetAverageTemperaturePerSeason(double expected, double actual) {
+        Assertions.assertEquals(expected, actual);
+    }
+}
+
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/enum_types/TestEnum.java:42]: "Violation: Useless blank lines
Solution: Remove blank lines"

**Review by rakovets — CHANGES_REQUESTED** [review]
need to fix

**Pull request closed without merging**